### PR TITLE
LOK-2306: Send Alerts via eMail (Lokahi) 

### PR DESCRIPTION
**Pull request by aurang-zeb313**
## Description
As an IT Manager, I want to create a Monitoring Policy that delivers alerts and alert updates via e-mail to my most critical users.

We already have a mail-server container running on Tilt/Helm.  Enable it, and provide instructions on how to use it so that we can integrate it into a user’s mail environment.  sendmail or similar is fine.  Document clearly.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2306

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java [hunk 1]
```diff
@@ -336,7 +339,7 @@ private void updateData(MonitorPolicy policy, String tenantId, Boolean enabled)
             });
         });
         policy.getTags().forEach(tag -> tag.setTenantId(tenantId));
-        policy.setEnabled(enabled);
+        policy.setEnabled(true);
```
Comment (fooker) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java:342]: Is this an accident? If not, remove the parameter as it is unused.

File: alert/src/test/java/org/opennms/horizon/alertservice/grpc/MonitoringPolicyGrpcTest.java [hunk 2]
```diff
@@ -131,7 +134,8 @@ void prepareTest() throws VerificationException, IOException {
                 mockAlertDefinitionRepo,
                 mockAlertRepository,
                 mockTagRepository,
-                mockTagOperationProducer));
+                mockTagOperationProducer,
```
Comment (fooker) [inline alert/src/test/java/org/opennms/horizon/alertservice/grpc/MonitoringPolicyGrpcTest.java:137]: fmt

File: notifications/src/main/java/org/opennms/horizon/notifications/api/email/SmtpEmailAPI.java [hunk 3]
```diff
@@ -49,13 +49,11 @@ public void sendEmail(String emailAddress, String subject, String bodyHtml) thro
             try {
                 MimeMessage mimeMessage = sender.createMimeMessage();
                 MimeMessageHelper helper = new MimeMessageHelper(mimeMessage);
-
                 helper.setTo(emailAddress);
                 helper.setFrom(fromAddress);
 
                 helper.setSubject(subject);
                 helper.setText(bodyHtml, true);
-
```
Comment (fooker) [inline notifications/src/main/java/org/opennms/horizon/notifications/api/email/SmtpEmailAPI.java:58]: Why?

File: notifications/src/main/java/org/opennms/horizon/notifications/config/EmailConfig.java [hunk 4]
```diff
@@ -32,6 +32,7 @@
 import org.opennms.horizon.notifications.api.email.EmailAPI;
 import org.opennms.horizon.notifications.api.email.SmtpEmailAPI;
 import org.opennms.horizon.notifications.exceptions.NotificationAPIRetryableException;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (fooker) [inline notifications/src/main/java/org/opennms/horizon/notifications/config/EmailConfig.java:35]: Revert

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java [hunk 5]
```diff
@@ -36,6 +36,7 @@
 import org.opennms.horizon.notifications.model.MonitoringPolicy;
 import org.opennms.horizon.notifications.repository.MonitoringPolicyRepository;
 import org.opennms.horizon.notifications.tenant.WithTenant;
+import org.springframework.beans.factory.annotation.Value;
```
Comment (fooker) [inline notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java:39]: Revert

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java [hunk 6]
```diff
@@ -106,6 +107,7 @@ public void postNotification(Alert alert) {
         if (notifyEmail) {
             postEmailNotification(alert);
         }
+
```
Comment (fooker) [inline notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java:110]: Revert

File: notifications/src/main/resources/application.yml [hunk 7]
```diff
@@ -59,3 +59,4 @@ management:
       probes:
         enabled: true
       show-details: always
+
```
Comment (fooker) [inline notifications/src/main/resources/application.yml:62]: Revert

File: notifications/src/test/java/org/opennms/horizon/notifications/service/NotificationServiceImplTest.java [hunk 8]
```diff
@@ -84,8 +86,10 @@ public void testNotificationPagerDutyPolicy() throws NotificationException {
         monitoringPolicy.setId(1);
         monitoringPolicy.setTenantId("T1");
         monitoringPolicy.setNotifyByPagerDuty(true);
+        List<MonitoringPolicy> policies = new ArrayList<>();
+        policies.add(monitoringPolicy);
         Mockito.when(monitoringPolicyRepository.findByTenantIdAndIdIn("T1", List.of(1L)))
-                .thenReturn(List.of(monitoringPolicy));
+                .thenReturn(policies);
```
Comment (fooker) [inline notifications/src/test/java/org/opennms/horizon/notifications/service/NotificationServiceImplTest.java:92]: Why altering this? If the list needs to be mutable for any reason, there are better ways to do this.

**Review by fooker — CHANGES_REQUESTED** [review]

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java (hunk 1 above)
Comment (aurang-zeb313) [reply alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java]: Done

File: alert/src/test/java/org/opennms/horizon/alertservice/grpc/MonitoringPolicyGrpcTest.java (hunk 2 above)
Comment (aurang-zeb313) [reply alert/src/test/java/org/opennms/horizon/alertservice/grpc/MonitoringPolicyGrpcTest.java:137]: Done

File: notifications/src/main/resources/application.yml (hunk 7 above)
Comment (aurang-zeb313) [reply notifications/src/main/resources/application.yml:62]: run spotless  and its reverted.

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java (hunk 6 above)
Comment (aurang-zeb313) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java]: reverted

File: notifications/src/main/java/org/opennms/horizon/notifications/api/email/SmtpEmailAPI.java (hunk 3 above)
Comment (aurang-zeb313) [reply notifications/src/main/java/org/opennms/horizon/notifications/api/email/SmtpEmailAPI.java:58]: reverted. run spotless

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java (hunk 5 above)
Comment (aurang-zeb313) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationService.java]: run spotless and reverted

**Review by fooker — APPROVED** [review]

File: charts/lokahi/values.yaml
Comment (mershad-manesh) [inline charts/lokahi/values.yaml:1]: This change is not correct!
We shouldn't hardcode these values nor use this email server for development purpose

**Pull request merged**